### PR TITLE
Update org name on login

### PR DESCRIPTION
**Pull request by qwerty287**
and create a new org once a user is created (logged in for the first time)

closes https://github.com/woodpecker-ci/woodpecker/issues/3342

### CODE REVIEW

**Comment by 6543** [comment]
I think we should start writing tests for this function :sweat_smile:

File: server/api/login.go [hunk 1]
```diff
@@ -119,6 +119,34 @@ func HandleAuth(c *gin.Context) {
...
+				_ = c.AbortWithError(http.StatusInternalServerError, err)
+				return
+			}
+			org = &model.Org{
+				Name:    u.Login,
+				IsUser:  true,
+				Private: false,
+			}
+			if err := _store.OrgCreate(org); err != nil {
+				log.Error().Err(err).Msgf("on user creation, could not mark org as user")
```
Comment (anbraten) [inline server/api/login.go:133]:
```suggestion
				log.Error().Err(err).Msgf("on user creation, could create org for user")
```

File: server/api/login.go [hunk 2]
```diff
@@ -119,6 +119,34 @@ func HandleAuth(c *gin.Context) {
...
 			}
+		} else {
+			if err != nil && !errors.Is(err, types.RecordNotExist) {
+				_ = c.AbortWithError(http.StatusInternalServerError, err)
+				return
+			}
+			org = &model.Org{
+				Name:    u.Login,
+				IsUser:  true,
+				Private: false,
```
Comment (anbraten) [inline server/api/login.go:130]: What does a public org allow? List all public repos?

File: server/api/login.go [hunk 3]
```diff
@@ -119,6 +119,34 @@ func HandleAuth(c *gin.Context) {
 			if err := _store.OrgUpdate(org); err != nil {
```
Comment (anbraten) [inline server/api/login.go:119]:
```suggestion
      u.OrgID = org.ID
			if err := _store.OrgUpdate(org); err != nil {
```

File: server/api/login.go [hunk 4]
```diff
@@ -119,6 +119,34 @@ func HandleAuth(c *gin.Context) {
...
+			if err := _store.OrgCreate(org); err != nil {
+				log.Error().Err(err).Msgf("on user creation, could not mark org as user")
+			}
+			u.OrgID = org.ID
+		}
+	}
+
+	// update org name
+	if u.Login != tmpuser.Login {
+		org, err := _store.OrgGet(u.OrgID)
```
Comment (anbraten) [inline server/api/login.go:141]: Can org-id be null / `0` for users created before we added the user-orgs?

**Comment by 6543** [comment]
> That's really hard to do as it requires a lot of mocking. It needs the gin context with store and a forge.

we have a mock for both already ... anyway this is not blocking, just a todo to be done

File: server/api/login.go [hunk 5]
```diff
@@ -116,9 +116,38 @@ func HandleAuth(c *gin.Context) {
...
+		if err != nil {
+			log.Error().Err(err).Msgf("cannot get org %s", u.Login)
+			c.Redirect(http.StatusSeeOther, server.Config.Server.RootPath+"/login?error=internal_error")
+			return
+		}
+		org.Name = u.Login
```
Comment (6543) [inline server/api/login.go:148]: if org.Name != u.Login ... onyl then update

**Review by 6543 — APPROVED** [review]

**Pull request merged**